lib/arborist/load-virtual.js

```diff
@@ -71,14 +79,86 @@ module.exports = cls => class VirtualLoader extends cls {
     root.optional = false
     root.devOptional = false
     root.peer = false
+    this[checkRootEdges](s, root)
     s.add(root)
     this.virtualTree = root
     const {links, nodes} = this[resolveNodes](s, root)
     await this[resolveLinks](links, nodes)
     this[assignParentage](nodes)
+    if (this[flagsSuspect])
+      this[reCalcDepFlags]()
     return root
   }
 
+  [reCalcDepFlags] () {
+    // reset all dep flags
+    for (const node of this.virtualTree.inventory.values()) {
+      node.extraneous = true
+      node.dev = true
+      node.optional = true
+      node.devOptional = true
+      node.peer = true
+    }
+    calcDepFlags(this.virtualTree, true)
+  }
+
+  // check the lockfile deps, and see if they match.  if they do not
+  // then we have to reset dep flags at the end.  for example, if the
```

Comment:
Maybe we can fix this comment and capitalize  `For`, and remove the extra space in front.

Reply:
hahah, no thanks.  if we start requiring that all code comments be grammatically correct, I'll have a _lot_ of editing to do.

Reply:
Haha, no worries 😂 Just something I spotted.